internal/pkg/cosign/fulcio/fulcio.go

```diff
@@ -12,30 +12,31 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package cosign
+package fulcio
 
 import (
 	"context"
 	"crypto"
 	"io"
 
+	"github.com/sigstore/cosign/internal/pkg/cosign"
 	"github.com/sigstore/cosign/pkg/oci"
 	"github.com/sigstore/cosign/pkg/oci/static"
 )
 
-// FulcioSignerWrapper still needs to actually upload keys to Fulcio and receive
+// SignerWrapper still needs to actually upload keys to Fulcio and receive
 // the resulting `Cert` and `Chain`, which are added to the returned `oci.Signature`
-type FulcioSignerWrapper struct {
-	Inner Signer
+type SignerWrapper struct {
+	inner cosign.Signer
 
-	Cert, Chain []byte
+	cert, chain []byte
 }
 
-var _ Signer = (*FulcioSignerWrapper)(nil)
+var _ cosign.Signer = (*SignerWrapper)(nil)
 
-// Sign implements `Signer`
-func (fs *FulcioSignerWrapper) Sign(ctx context.Context, payload io.Reader) (oci.Signature, crypto.PublicKey, error) {
-	sig, pub, err := fs.Inner.Sign(ctx, payload)
+// Sign implements `cosign.Signer`
+func (fs *SignerWrapper) Sign(ctx context.Context, payload io.Reader) (oci.Signature, crypto.PublicKey, error) {
+	sig, pub, err := fs.inner.Sign(ctx, payload)
```

Comment:
I don't really grok why this is wrapping *any* Signer instead of having a specific Signer implementation embedded here.  Can this truly be composed with other Signer implementations arbitrarily?  What other compositions make sense?

Reply:
not quite arbitrarily but we made the `Signer` abstraction for a reason. I think noting the requirements for our public instance in comments would be a legitimate way of documenting them (vs. using the type system)